src/backend/web.rs

```diff
@@ -800,7 +799,7 @@ fn map_map_mode(mode: crate::MapMode) -> u32 {
 }
 
 type JsFutureResult = Result<wasm_bindgen::JsValue, wasm_bindgen::JsValue>;
-type FutureMap<T> = futures::future::Map<wasm_bindgen_futures::JsFuture, fn(JsFutureResult) -> T>;
+type MakeBoxedSendFuture<T> = MakeSendFuture<std::pin::Pin<Box<dyn Future<Output = T>>>>;
```

Comment:
Does this allocate now? Could we keep the old signature somehow?

Reply:
It does allocate now. I have to use `Box<dyn Future<...>>` to return an `async` block. Which I am only doing because it's nicer to use than creating a concrete type that implements `Future`. This is definitely something than can be improved though.

Reply:
@grovesNL I copy-pasted `FuturesExt::Map` to get this working. Fair tradeoff, I suppose.